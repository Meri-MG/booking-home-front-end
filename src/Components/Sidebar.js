import { useState } from 'react';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { useNavigate } from 'react-router';
import { AiOutlineClose, AiOutlineMenu } from 'react-icons/ai';
import { logUserOut } from '../Redux/Login/Login';
import logo from '../Assets/logo.jpeg';

const Sidebar = () => {
  const [showSidebar, setShowSidebar] = useState(false);
  const dispatch = useDispatch();
  const navigateTo = useNavigate();
  const signUserOut = () => {
    dispatch(logUserOut()).then(() => navigateTo('/Login'));
    setShowSidebar(!showSidebar);
  };
  return (
    <>
      <div className="flex flex-col items-center justify-center">
        {showSidebar ? (
          <button
            className="hover:ring-2 hover:ring-[#555555]-600 outline-0 fixed left-10 top-6 z-50 border-solid border-2 border-[#555555]-600 rounded-full flex text-2xl text-[#555555] p-2 items-center cursor-pointer"
            onClick={() => setShowSidebar(!showSidebar)}
            type="button"
          >
            <AiOutlineClose />
          </button>
        ) : (
          <button
            onClick={() => setShowSidebar(!showSidebar)}
            className="hover:ring-2 hover:ring-[#555555]-600 outline-0 fixed left-10 top-6  z-30 border-solid border-2 border-[#555555]-600 flex items-center text-[#555555] text-2xl rounded-md p-2 cursor-pointer"
            type="button"
          >
            <AiOutlineMenu />
          </button>
        )}

        <div
          className={`h-screen top-0 left-0 w-80 bg-[#fdfcf7] p-10 pl-10 px-5 fixed transition-ease-in-out ease-in-out duration-300 z-30 ${
            showSidebar ? 'translate-x-0 ' : '-translate-x-full'
          }`}
        >
          <nav>
            <div className="container flex flex-wrap justify-between items-center mx-auto">
              <div className="px-6 pt-8 pb-4">
                <Link to="/">
                  <button
                    className="uppercase"
                    type="button"
                    onClick={() => setShowSidebar(!showSidebar)}
                  >
                    <img
                      src={logo}
                      alt="logo"
                      className="w-60 h-15 rounded-full"
                    />
                  </button>
                </Link>
              </div>
              <div>
                <ul>
                  <li>
                    <Link
                      className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                      to="/"
                    >
                      <button
                        className="uppercase"
                        type="button"
                        onClick={() => setShowSidebar(!showSidebar)}
                      >
                        Apartments
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                      to="/reservations"
                    >
                      <button
                        className="uppercase"
                        type="button"
                        onClick={() => setShowSidebar(!showSidebar)}
                      >
                        My Reservations
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                      to="/addApartment"
                    >
                      <button
                        className="uppercase"
                        type="button"
                        onClick={() => setShowSidebar(!showSidebar)}
                      >
                        Add Apartment
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                      to="/deleteApartment"
                    >
                      <button
                        className="uppercase"
                        type="button"
                        onClick={() => setShowSidebar(!showSidebar)}
                      >
                        Delete Apartment
                      </button>
                    </Link>
                  </li>
                  <li>
                    <Link
                      className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                      to="/"
                    >
                      <button
                        className="uppercase"
                        type="button"
                        onClick={() => signUserOut()}
                      >
                        Log out
                      </button>
                    </Link>
                  </li>
                </ul>
              </div>
            </div>
          </nav>
          <div className="footer fixed inset-x-0 bottom-0 p-4">
            <div className="py-8 px-6 md:flex md:items-center md:justify-between ">
              <span className="text-sm text-[#555555] sm:text-center">
                © 2022
                <a href="https://www.microverse.org"> Microverse™</a>
                . All Rights
                Reserved.
              </span>
            </div>
          </div>
        </div>
      </div>
    </>
  );
};

export default Sidebar;
