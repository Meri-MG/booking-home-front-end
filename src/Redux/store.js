import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';
import loginReducer from './Login/Login';
import registerReducer from './Register/Register';
import apartmentsReducer from './Apartments/Apartments';
import apartmentsDetailsReducer from './Details/Details';
import reservationsReducer from './Reserved/Reserved';

const reducers = combineReducers({
  LogIn: loginReducer,
  Register: registerReducer,
  Apartments: apartmentsReducer,
  Details: apartmentsDetailsReducer,
  Reservations: reservationsReducer,
});

const middlewares = [thunk, logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
