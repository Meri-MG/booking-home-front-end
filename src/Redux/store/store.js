import { applyMiddleware, combineReducers, createStore } from 'redux';
import logger from 'redux-logger';
import thunk from 'redux-thunk';

const reducers = combineReducers({
});

const middlewares = [thunk,logger];

const store = createStore(reducers, applyMiddleware(...middlewares));

export default store;
