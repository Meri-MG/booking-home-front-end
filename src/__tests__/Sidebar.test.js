import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/store';
import Sidebar from '../Components/Sidebar';

it('renders sidebar correctly', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Sidebar />
        </Router>
      </Provider>,
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
