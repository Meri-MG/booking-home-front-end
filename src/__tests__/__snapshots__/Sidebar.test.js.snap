// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders sidebar correctly 1`] = `
<div
  className="flex flex-col items-center justify-center"
>
  <button
    className="hover:ring-2 hover:ring-[#555555]-600 outline-0 fixed left-10 top-6  z-30 border-solid border-2 border-[#555555]-600 flex items-center text-[#555555] text-2xl rounded-md p-2 cursor-pointer"
    onClick={[Function]}
    type="button"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M904 160H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0 624H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8zm0-312H120c-4.4 0-8 3.6-8 8v64c0 4.4 3.6 8 8 8h784c4.4 0 8-3.6 8-8v-64c0-4.4-3.6-8-8-8z"
      />
    </svg>
  </button>
  <div
    className="h-screen top-0 left-0 w-80 bg-[#fdfcf7] p-10 pl-10 px-5 fixed transition-ease-in-out ease-in-out duration-300 z-30 -translate-x-full"
  >
    <nav>
      <div
        className="container flex flex-wrap justify-between items-center mx-auto"
      >
        <div
          className="px-6 pt-8 pb-4"
        >
          <a
            href="/"
            onClick={[Function]}
          >
            <button
              className="uppercase"
              onClick={[Function]}
              type="button"
            >
              <img
                alt="logo"
                className="w-60 h-15 rounded-full"
                src="logo.jpeg"
              />
            </button>
          </a>
        </div>
        <div>
          <ul>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  Apartments
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/reservations"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  My Reservations
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/addApartment"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  Add Apartment
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/deleteApartment"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  Delete Apartment
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/rentals"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  For Rent
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/markets"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  For sale
                </button>
              </a>
            </li>
            <li>
              <a
                className="block py-2 px-6 text-md text-[#555555]-700 hover:bg-[#96bf01] hover:text-[#fdfcf7] hover:rounded-md uppercase border-b-4"
                href="/"
                onClick={[Function]}
              >
                <button
                  className="uppercase"
                  onClick={[Function]}
                  type="button"
                >
                  Log out
                </button>
              </a>
            </li>
          </ul>
        </div>
      </div>
    </nav>
    <div
      className="footer fixed inset-x-0 bottom-0 p-4"
    >
      <div
        className="py-8 px-6 md:flex md:items-center md:justify-between "
      >
        <span
          className="text-sm text-[#555555] sm:text-center"
        >
          © 2022
          <a
            href="https://www.microverse.org"
          >
             Microverse™
          </a>
          . All Rights Reserved.
        </span>
      </div>
    </div>
  </div>
</div>
`;
